contracts/strategies/QVSimple.sol

```diff
@@ -0,0 +1,270 @@
+
+    /// ===============================
+    /// ========= Initialize ==========
+    /// ===============================
+
+    /// @notice Initialize the strategy
+    /// @param _poolId The pool id
+    /// @param _data The data to initialize the strategy (Must include RecipientInitializeData and QVSimpleInitializeData)
+    function initialize(uint256 _poolId, bytes memory _data) external override {
+        __BaseStrategy_init(_poolId);
+
+        (
+            IRecipientsExtension.RecipientInitializeData memory recipientInitializeData,
+            QVSimpleInitializeData memory qvSimpleInitializeData
+        ) = abi.decode(_data, (IRecipientsExtension.RecipientInitializeData, QVSimpleInitializeData));
+
+        __RecipientsExtension_init(recipientInitializeData);
+
+        maxVoiceCreditsPerAllocator = qvSimpleInitializeData.maxVoiceCreditsPerAllocator;
+        allocationStartTime = qvSimpleInitializeData.allocationStartTime;
+        allocationEndTime = qvSimpleInitializeData.allocationEndTime;
+
+        emit Initialized(_poolId, _data);
+    }
+
+    /// ======================
+    /// ======= Events =======
+    /// ======================
+
+    /// @notice Emitted when an allocator is added
+    /// @param allocator The allocator address
+    /// @param sender The sender of the transaction
+    event AllocatorAdded(address indexed allocator, address sender);
+
+    /// @notice Emitted when an allocator is removed
+    /// @param allocator The allocator address
+    /// @param sender The sender of the transaction
+    event AllocatorRemoved(address indexed allocator, address sender);
+
+    /// ======================
+    /// ======= Struct =======
+    /// ======================
+
+    /// @notice The parameters used to initialize the strategy
+    struct QVSimpleInitializeData {
+        uint64 allocationStartTime;
+        uint64 allocationEndTime;
+        uint256 maxVoiceCreditsPerAllocator;
+    }
+
+    /// ====================================
+    /// ==== External/Public Functions =====
+    /// ====================================
+
+    /// @notice Add allocator
+    /// @dev Only the pool manager(s) can call this function and emits an `AllocatorAdded` event
+    /// @param _allocator The allocator address
+    function addAllocator(address _allocator) external onlyPoolManager(msg.sender) {
+        allowedAllocators[_allocator] = true;
+
+        emit AllocatorAdded(_allocator, msg.sender);
+    }
+
+    /// @notice Remove allocator
+    /// @dev Only the pool manager(s) can call this function and emits an `AllocatorRemoved` event
+    /// @param _allocator The allocator address
+    function removeAllocator(address _allocator) external onlyPoolManager(msg.sender) {
+        allowedAllocators[_allocator] = false;
+
+        emit AllocatorRemoved(_allocator, msg.sender);
+    }
+
+    /// ================================
+    /// ========== Modifier ============
```

Comment:
lets move the modifiers above the external public functions to follow the style guide